src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1092,21 +1127,18 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
   }
   // lie: use transient failure from the transport to indicate goaway has been
   // received.
-  connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, status,
-                         "got_goaway");
+  if (!grpc_core::test_only_disable_transient_failure_state_notification) {
+    connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, status,
+                           "got_goaway");
+  }
 }
 
 static void maybe_start_some_streams(grpc_chttp2_transport* t) {
   grpc_chttp2_stream* s;
-  // cancel out streams that haven't yet started if we have received a GOAWAY
+  // maybe cancel out streams that haven't yet started if we have received a
+  // GOAWAY
   if (t->goaway_error != GRPC_ERROR_NONE) {
-    while (grpc_chttp2_list_pop_waiting_for_concurrency(t, &s)) {
-      grpc_chttp2_cancel_stream(
-          t, s,
-          grpc_error_set_int(
-              GRPC_ERROR_CREATE_FROM_STATIC_STRING("GOAWAY received"),
-              GRPC_ERROR_INT_GRPC_STATUS, GRPC_STATUS_UNAVAILABLE));
```

Comment:
It looks like we're no longer setting the status to UNAVAILABLE here, but I see that the tests are still checking for this, and they are presumably passing, so it must be happening somewhere else.  Where is this happening now?

Reply:
https://github.com/grpc/grpc/blob/cf81e41162603d229d0c8ef05f137a0afbe1f3c6/src/core/ext/transport/chttp2/transport/chttp2_transport.cc#L1054

goaway_error sets the status already

Reply:
Got it, thanks!